spec_integration/queue_runner.rb

```diff
@@ -0,0 +1,75 @@
+require 'knapsack_pro'
+require 'json'
+
+ENV['KNAPSACK_PRO_CI_NODE_BUILD_ID'] = SecureRandom.uuid
+ENV['KNAPSACK_PRO_TEST_SUITE_TOKEN_RSPEC'] = 'fake-api-token'
+ENV['KNAPSACK_PRO_TEST_DIR'] = 'spec_integration'
+ENV['KNAPSACK_PRO_TEST_FILE_PATTERN'] = "spec_integration/**{,/*/**}/*_spec.rb"
+
+RSPEC_OPTIONS = ENV.fetch('TEST__RSPEC_OPTIONS')
+SHOW_DEBUG_LOG = ENV['TEST__SHOW_DEBUG_LOG'] == 'true'
+BATCHED_TESTS = JSON.load(ENV.fetch('TEST__BATCHED_TESTS'))
+
+module KnapsackProExtensions
+  module QueueAllocatorExtension
+    def test_file_paths(can_initialize_queue, executed_test_files)
+      @@batch_index ||= 0
```

Comment:
An ivar should be enough:

```suggestion
      @batch_index ||= 0
```